benchmark/rocblas/blas3/gemm.cpp

```diff
@@ -0,0 +1,214 @@
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename gemm.cpp
+ *
+ **************************************************************************/
+
+#include "../utils.hpp"
+
+template <typename scalar_t>
+std::string get_name(std::string t_a, std::string t_b, int m, int k, int n) {
+  std::ostringstream str{};
+  str << "BM_Gemm<" << blas_benchmark::utils::get_type_name<scalar_t>() << ">/"
+      << t_a << "/" << t_b << "/" << m << "/" << k << "/" << n;
+  return str.str();
+}
+
+template <typename scalar_t, typename... args_t>
+static inline void rocblas_gemm_f(args_t&&... args) {
+  if constexpr (std::is_same_v<scalar_t, float>) {
+    CHECK_ROCBLAS_STATUS(rocblas_sgemm(std::forward<args_t>(args)...));
+  } else if constexpr (std::is_same_v<scalar_t, double>) {
+    CHECK_ROCBLAS_STATUS(rocblas_dgemm(std::forward<args_t>(args)...));
+  }
+  return;
+}
+
+template <typename scalar_t>
+void run(benchmark::State& state, rocblas_handle& rb_handle, int t_a_i,
+         int t_b_i, index_t m, index_t k, index_t n, scalar_t alpha,
+         scalar_t beta, bool* success) {
+  // Standard test setup.
+  std::string t_a = blas_benchmark::utils::from_transpose_enum(
+      static_cast<blas_benchmark::utils::Transposition>(t_a_i));
+  std::string t_b = blas_benchmark::utils::from_transpose_enum(
+      static_cast<blas_benchmark::utils::Transposition>(t_b_i));
+  const char* t_a_str = t_a.c_str();
+  const char* t_b_str = t_b.c_str();
+
+  index_t lda = t_a_str[0] == 'n' ? m : k;
+  index_t ldb = t_b_str[0] == 'n' ? k : n;
+  index_t ldc = m;
+
+  {
+    // The counters are double. We convert m, n and k to double to avoid
+    // integer overflows for n_fl_ops and bytes_processed
+    double m_d = static_cast<double>(m);
+    double n_d = static_cast<double>(n);
+    double k_d = static_cast<double>(k);
+
+    state.counters["m"] = m_d;
+    state.counters["k"] = k_d;
+    state.counters["n"] = n_d;
+
+    double mem_readA = m_d * k_d;
+    double mem_readB = k_d * n_d;
+    double mem_writeC = m_d * n_d;
+    double mem_readC = (beta != scalar_t{0}) ? m_d * n_d : 0;
+    double total_mem =
+        (mem_readA + mem_readB + mem_readC + mem_writeC) * sizeof(scalar_t);
+    state.counters["bytes_processed"] = total_mem;
+    state.SetBytesProcessed(state.iterations() * total_mem);
+
+    double nflops_AtimesB = (2 * k_d) * m_d * n_d;
+    double nflops_addBetaC = (beta != scalar_t{0}) ? 2 * m_d * n_d : 0;
+    double nflops_tot = nflops_AtimesB + nflops_addBetaC;
+    state.counters["n_fl_ops"] = nflops_tot;
+    state.SetItemsProcessed(state.iterations() * nflops_tot);
```

Comment:
Did you check if using `state.iterations()` here work? I think this and *line 80* should be moved after the loop *line 168-175*, otherwise the number of iteration could be 0

Reply:
Correct! I've just checked the doc and the results. Will fix these thank you.